contracts/ERC721A.sol

```diff
@@ -62,19 +75,43 @@ contract ERC721A is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
         _symbol = symbol_;
     }
 
+    /**
+     * @dev Skips the zero index.
+     * This method must be called before any mints (e.g. in the consturctor).
+     */
+    function _initOneIndexed() internal {
+        require(!_exists(0), "ERC721A: 0 index already occupied.");
+        currentIndex = 1;
+        totalBurned = 1;
+        _ownerships[0].burned = true;
+    }
+
     /**
      * @dev See {IERC721Enumerable-totalSupply}.
      */
     function totalSupply() public view override returns (uint256) {
-        return currentIndex;
+        return currentIndex - totalBurned;
     }
 
     /**
      * @dev See {IERC721Enumerable-tokenByIndex}.
+     * This read function is O(totalSupply). If calling from a separate contract, be sure to test gas first.
+     * It may also degrade with extremely large collection sizes (e.g >> 10000), test for your use case.
      */
     function tokenByIndex(uint256 index) public view override returns (uint256) {
-        require(index < totalSupply(), 'ERC721A: global index out of bounds');
-        return index;
+        uint256 numMintedSoFar = currentIndex;
+        uint256 tokenIdsIdx = 0;
+        for (uint256 i = 0; i < numMintedSoFar; i++) {
+            TokenOwnership memory ownership = _ownerships[i];
+            if (!ownership.burned) {
+                if (tokenIdsIdx == index) {
+                    return i;
+                }
+                tokenIdsIdx++;
+            }
+        }
+        require(false, 'ERC721A: global index out of bounds');
```

Comment:
you can use just `revert('ERC721A: global index out of bounds')`

Reply:
Updated.

https://github.com/Vectorized/ERC721A/blob/feature/burnable/contracts/ERC721A.sol#L113